### PR TITLE
[10.0] purchase_request_to_rfq: several improvements from v9

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Continue with the work started in https://github.com/OCA/purchase-workflow/pull/287 making several proposed improvements and porting changes made in v9 of the module.

* [x] : Changes proposed by @sylvain-garancher in https://github.com/OCA/purchase-workflow/pull/287.
* [x] : Cherry-pick 93d284bad472623ffdd3aa34bb0482ee8711e7bb
* [x] : Fixes done in https://github.com/OCA/purchase-workflow/pull/350 (merged)
* [x] : Fix done in https://github.com/OCA/purchase-workflow/pull/371 (merged)(cherry-pick 1458c34aa31dd810fbcf4306c1e86ead4f5ccd22)
* [x] : Fix proposed in https://github.com/OCA/purchase-workflow/pull/375 (merged)
* [x] : Changes proposed in https://github.com/OCA/purchase-workflow/pull/395 (merged)
* [x] : Add changes from https://github.com/OCA/purchase-workflow/pull/340 (merged)
* [x] : Add https://github.com/OCA/purchase-workflow/pull/399 (merged)

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
Ready to review. 

cc @adrienpeiffer @jbeficent @mreficent @sylvain-garancher @sergio-teruel @aheficent @sbidoul @aduijs

**Review by MiquelRForgeFlow — APPROVED** [review]
I give you a :1st_place_medal: 

:+1:

File: purchase_request_to_rfq/models/purchase_request.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class PurchaseRequestLine(models.Model):
     @api.depends('purchase_lines')
     def _compute_is_editable(self):
         super(PurchaseRequestLine, self)._compute_is_editable()
-        for rec in self.filtered(lambda p: p in self and p.purchase_lines):
+        for rec in self.mapped('purchase_lines'):
```
Comment (Garamotte) [inline purchase_request_to_rfq/models/purchase_request.py]: `self.filtered(lambda p: p in self and p.purchase_lines)` returns all `purchase.request.line` that are in `self` and have at least one linked `purchase.order.line`.
`self.mapped('purchase_lines')` returns all `purchase.order.line` linked to every `purchase.request.line` in `self`.

The right modification is : `self.filtered(lambda p: p.purchase_lines)`

**Review by Garamotte — CHANGES_REQUESTED** [review]
I think you can also include https://github.com/OCA/purchase-workflow/pull/340

File: purchase_request_to_rfq/models/purchase_request.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply purchase_request_to_rfq/models/purchase_request.py]: @sylvain-garancher You are totally right, I'll fix it

**Review by gmeficent — APPROVED** [review]
395 and 399 merged.

Pending to discuss the flow.

**Comment by LoisRForgeFlow** [comment]
This is ready to review. Can someone add the tag?

**Comment by LoisRForgeFlow** [comment]
@sylvain-garancher can you update your review?

**Comment by LoisRForgeFlow** [comment]
All changes are merged in v9, this is ready.

**Review by Garamotte — APPROVED** [review]
LGTM :+1:

**Comment by JordiBForgeFlow** [comment]
@adrienpeiffer @sbidoul your review is welcome!

**Comment by LoisRForgeFlow** [comment]
I think it is time to move this forward, since we are introducing some new changes. V10 branch cannot be so behind.

**Comment by JordiBForgeFlow** [comment]
@benwillig Do you agree?

**Comment by LoisRForgeFlow** [comment]
Ping - Can we move this forward?

**Review by benwillig — APPROVED** [review]
LGTM :+1:

**Pull request merged**